### PR TITLE
Holocene: eip-1559 parameters in extraData

### PR DESCRIPTION
**Pull request by roberto-bayardo**
**Description**

EIP-1559 parameters need to be in the ExecutionPayload, not just the PayloadAttributes, to allow geth to properly include the encoded parameters in the block header.  So instead of using nonce, we use the extraData field which is already in the ExecutionPayload. This will also reserve nonce to be used for sub-second block times.

### CODE REVIEW

File: specs/protocol/holocene/exec-engine.md
Comment (tynes) [inline specs/protocol/holocene/exec-engine.md:100]: Does this mean there needs to be a new gossipsub topic for the modified execution payload? For unsafe blocks. Each version gets its own topic, see https://github.com/ethereum-optimism/specs/pull/396 which modifies the execution payload and a new topic is created

Comment (roberto-bayardo) [reply specs/protocol/holocene/exec-engine.md:100]: Hmm, since it is just adding a new optional field and the block itself is not changed, I assumed no.  But I am not so familiar with how the p2p layer does validation and may have to dig deeper to confirm.

**Comment by roberto-bayardo** [comment]
As discussed in discord, updated spec to use extraData instead of nonce.

File: specs/protocol/holocene/exec-engine.md
Comment (protolambda) [inline specs/protocol/holocene/exec-engine.md]:
```suggestion
  - if Holocene is not active in `parent_header.timestamp`, then `parent_header.extraData` may be set to anything valid in pre-Holocene context. The  [canyon base fee parameter constants](../exec-engine.md#1559-parameters), or prior EIP-1559 constants are used for the block's base fee parameters.
    - While not set in regular pre-Holocene blocks, the `extraData` was set in the genesis block of some pre-Holocene chains, and other blocks with pre-Bedrock legacy content. It is thus important to check the hardfork before parsing the `extraData`.
  - if Holocene is active in `parent_header.timestamp`, then `parent_header.extraData` is non-empty, with the following encoding:
    - the first byte is set to `0`, used as version byte for the content.
    - the next 8 bytes are used to encode the 1559 params following the [`eip1559Params`
encoding](#eip1559params-encoding).
    - the `extraData` ends there, at 9 bytes length. The `extraData` has a maximum capacity of 32 bytes (to fit in the L1 beacon-chain `extraData` data-type), and may be modified/extended in future versions.
```

A hardfork should also not retro-actively apply constraints on prior blocks, like the parent-block.
The spec needs to call out that the transition block does not read the 1559 params from the more arbitrary pre-holocene block. 
The previous spec rule would create weird behavior for the block right after genesis, if Holocene wasn't active then yet.

File: specs/protocol/holocene/system-config.md
Comment (protolambda) [inline specs/protocol/holocene/system-config.md:33]: It may be worth clarifying that this is the format emitted by the `SystemConfig`. Not the same as the format in the block-header `extraData`.

Comment (roberto-bayardo) [reply specs/protocol/holocene/system-config.md:33]: done

File: specs/protocol/holocene/exec-engine.md
Comment (roberto-bayardo) [reply specs/protocol/holocene/exec-engine.md]: thanks, did not realize it could be non-empty in previous cases. updated with some tweaking to your language.

**Review by clabby — APPROVED** [review]
lgtm. If we want, we can also take this opportunity to address your nitpick in the encoding: `denominator, elasticity` -> `elasticity, denominator`.

**Comment by roberto-bayardo** [comment]
> lgtm. If we want, we can also take this opportunity to address your nitpick in the encoding: `denominator, elasticity` -> `elasticity, denominator`.

Thanks, given the geth impl is all using this order now, may as well leave it be given how minor this is.

**Review by tynes — APPROVED** [review]
Looks good to me, would should let @protolambda have a final say before merge

**Comment by tynes** [comment]
Just to clarify, no need to add a new p2p topic correct?

**Comment by roberto-bayardo** [comment]
> Just to clarify, no need to add a new p2p topic correct?

Correct, we are no longer changing ExecutionPayload since it already contains the ExtraData field.

**Comment by roberto-bayardo** [comment]
> Just to clarify, no need to add a new p2p topic correct?

Correct, we are no longer changing ExecutionPayload since it already contains the ExtraData field.

**Comment by roberto-bayardo** [comment]
Oh shoot I think I borked this PR trying to rebase.

**Review by protolambda — APPROVED** [review]

**Review by protolambda — APPROVED** [review]

**Pull request merged**
